src/components/scroll-monitor.tsx

```diff
@@ -0,0 +1,71 @@
+import * as React from 'react'
+import { findDOMNode } from 'react-dom'
+
+interface Props {
+  onPage: (Number) => void
+}
+
+class ScrollMonitor extends React.Component<Props, null> {
+  page = 1
+  threshold = 0
+  teardown: () => void
+  lastChild: HTMLElement = null
+
+  static defaultProps: Props = {
+    onPage: (page: Number) => {}
+  }
+
+  getElement() {
+    return findDOMNode(this)
+  }
+
+  subscribe() {
+    let element = this.getElement()
+
+    element.addEventListener('scroll', this.check, { passive: true })
+
+    this.teardown = () => element.removeEventListener('scroll', this.check)
+
+    this.check()
+  }
+
+  ignore() {
+    this.teardown()
+  }
+
+  componentDidMount() {
+    this.subscribe()
+  }
+
+  componentDidUpdate() {
+    this.subscribe()
+  }
+
+  componentWillUnmount() {
+    this.ignore()
+  }
+
+  check = () => {
+    let container = this.getElement()
+    let nextChild = container.querySelector('[data-scroll="true"]')
+
+    if (!nextChild || nextChild === this.lastChild) {
+      return
+    }
+
+    let outer = container.getBoundingClientRect()
+    let inner = nextChild.getBoundingClientRect()
+
+    if (inner.top < outer.bottom + outer.height * 0.5) {
```

Comment:
Is there a simpler and perhaps more performant way to do this using `scrollTop` and `scrollHeight`? The answer may be `no` but I wanted to ask.